scripts/Muon/GUI/ElementalAnalysis/PeriodicTable/periodic_table_presenter.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import print_function
+
+
+class PeriodicTablePresenter(object):
+    def __init__(self, view, model):
+        self.view = view
+        self.model = model
+        self.set_buttons()
+
+    @property
+    def peak_data(self):
+        return self.model.peak_data
+
+    @property
+    def selection(self):
+        return self.view.ptable.getSelection()
+
+    def set_buttons(self):
+        for el in self.view.ptable.elements:
+            if el.symbol in self.model.peak_data:
+                self.view.ptable.enableElementButton(el.symbol)
+            else:
+                self.view.ptable.disableElementButton(el.symbol)
+
+    def element_data(self, element):
+        try:
+            return self.model.peak_data[element]
```

Comment:
I think this will always pass, if there is no data we cannot click it

Reply:
Good point